--- src/ls.c	2018-06-23 20:12:51.000000000 -0600
+++ src/ls.c	2019-01-04 08:36:50.000000000 -0700
@@ -4829,6 +4827,10 @@
   if (print_scontext)
     len += 1 + (format == with_commas ? strlen (f->scontext) : scontext_width);

+  // MOD1 - accounts for icon + space so columns are displayed correctly
+  if (print_with_color)
+      len+=2;
+
   len += quote_name_width (f->name, filename_quoting_options, f->quoted);

   if (indicator_style != none)
